src/interface/mesh_data.hpp

```diff
@@ -172,6 +172,7 @@ class MeshData {
   }
 
   void ClearCaches() {
+    block_data_.clear();
```

Comment:
:+1: 

Worth noting this is not required for the history MR necessarily. It's a fix of an older bug, as discussed on the Matrix channel.